code/ATMOSPHERICS/components/trinary_devices/mixer.dm

```diff
@@ -6,37 +6,29 @@
 
 	name = "gas mixer"
 
-	var/target_pressure = ONE_ATMOSPHERE
+	target_pressure = ONE_ATMOSPHERE
 	var/node1_concentration = 0.5
 	var/node2_concentration = 0.5
 
 	//node 3 is the outlet, nodes 1 & 2 are intakes
 
+// So we can CtrlClick without triggering the anchored message.
+/obj/machinery/atmospherics/trinary/mixer/can_be_pulled(user, grab_state, force, show_message)
+	return FALSE
+
 /obj/machinery/atmospherics/trinary/mixer/CtrlClick(mob/living/user)
-	if(!istype(user) || user.incapacitated())
-		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
-		return
-	if(!in_range(src, user) && !issilicon(usr))
-		return
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	toggle()
+	if(can_use_shortcut(user))
+		toggle(user)
 	return ..()
 
-/obj/machinery/atmospherics/trinary/mixer/AICtrlClick()
-	toggle()
+/obj/machinery/atmospherics/trinary/mixer/AICtrlClick(mob/living/silicon/user)
+	toggle(user)
 	return ..()
 
 /obj/machinery/atmospherics/trinary/mixer/AltClick(mob/living/user)
-	if(!istype(user) || user.incapacitated())
-		to_chat(user, "<span class='warning'>You can't do that right now!</span>")
-		return
-	if(!in_range(src, user) && !issilicon(usr))
-		return
-	if(!ishuman(usr) && !issilicon(usr))
-		return
-	set_max()
-	return
+	if(can_use_shortcut(user))
+		set_max(user)
+	return ..()
```

Comment:
Some of the other machineries modified in this PR don't include this call to parent -- I'd probably include it as `/obj/machinery/atmospherics/AltClick()` handles vent crawling.

Reply:
I suppose but, should we call parent here even though the mixer, filter, and pumps don't allow you ventcrawl into them?

Reply:
If we go up the parents...

https://github.com/ParadiseSS13/Paradise/blob/1d891384364a64f99948658d5124c92cb6e547e7/code/ATMOSPHERICS/atmospherics.dm#L295-L299

https://github.com/ParadiseSS13/Paradise/blob/1d891384364a64f99948658d5124c92cb6e547e7/code/_onclick/click.dm#L334-L343

Turns out alt-clicking would just pass the call to the parent to the parent if not a valid vent crawl object, showing the turf contents on the stat panel. Probably don't want that as it would ruin the flow of the shortcut -- feel free to ignore my comment above.

Reply:
Yeah, and in fact I should probably remove this `..()`, don't know why I added it here, but nowhere else.